docs/src/pages/components/data-grid/pagination/pagination.md

```diff
@@ -4,65 +4,75 @@ title: Data Grid - Pagination
 
-## Auto size
+{{"demo": "pages/components/data-grid/pagination/PageSizeAuto.js", "bg": "inline"}}
 
-The `autoPageSize` prop allows to auto-scale the `pageSize` to match the container height and the max number of rows that can be displayed without a vertical scroll bar.
-By default, this feature is off.
+### Controlled page size
 
-{{"demo": "pages/components/data-grid/pagination/AutoPaginationGrid.js", "bg": "inline"}}
+Use the `pageSize` prop to control the size of the pages.
 
-## Server-side pagination
+You can use the `onPageSizeChange` to listen to changes to the page size and update the prop accordingly.
 
-By default, pagination works on the client-side.
-To switch it to server-side, set `paginationMode="server"`.
-You also need to set the `rowCount` prop so that the grid knows the total number of pages.
-Finally, you need to handle the `onPageChange` callback to load the rows for the corresponding page.
+{{"demo": "pages/components/data-grid/pagination/PageSizeControlled.js", "bg": "inline"}}
 
-{{"demo": "pages/components/data-grid/pagination/ServerPaginationGrid.js", "bg": "inline"}}
+## Controlled page
+
+Use the `page` prop to control the size of the pages.
+
+You can use the `onPageChange` to listen to changes to the page size and update the prop accordingly.
+
+{{"demo": "pages/components/data-grid/pagination/PageControlled.js", "bg": "inline"}}
+
+## Server-side pagination
+
+By default, the pagination is handled on the client.
+This means you have to give the rows of all pages to the grid.
+If your dataset is too big, and you only want to fetch the current page, you can use server-side pagination.
 
 **Note**: For more information regarding server-side pagination in combination with controlled selection check [here](/components/data-grid/selection/#usage-with-server-side-pagination)
 
-## Cursor-based pagination
+### Basic implementation
 
-You can adapt the pagination for your cursor-based pagination.
-To do so, you just have to keep track of the next cursor associated with each page you fetched.
+- Set the prop `paginationMode` to `server`
+- Provide a `rowCount` prop to let the grid know how many pages there is
+- Add a `onPageChange` callback ot load the rows when the page changes
 
-{{"demo": "pages/components/data-grid/pagination/CursorPaginationGrid.js", "bg": "inline"}}
+{{"demo": "pages/components/data-grid/pagination/ServerPaginationGrid.js", "bg": "inline"}}
 
-## Customization
+### Cursor implementation
 
-You can customize the rendering of the pagination in the footer following [the component section](/components/data-grid/components/#pagination) of the documentation.
+You can also handle servers with cursor-based pagination.
+To do so, you just have to keep track of the next cursor associated with each page you fetched.
 
-## Paginate > 100 rows [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
```

Comment:
Why remove this demo? We broke these two anchor links

<img width="604" alt="Screenshot 2021-12-31 at 19 38 26" src="https://user-images.githubusercontent.com/3165635/147835784-35ea5526-2617-42b6-9268-03feaae0fe67.png">

https://mui.com/components/data-grid/getting-started/#feature-comparison

<img width="483" alt="Screenshot 2021-12-31 at 19 39 06" src="https://user-images.githubusercontent.com/3165635/147835792-71eb8195-afa7-4e25-861f-4d0e0d98715b.png">

https://mui.com/pricing/

Reply:
... I should have checked if there was links

Because I don't see the point of having this kind of demo.
Users do not learn anything from then compared to a line "only the pro version can paginate with more than 100 rows"
We now have long doc pages and I think we should focus on demos that demonstrate something.

Reply:
> I should have looked for links

@flaviendelangle I think that it's very hard to do in practice. I bet that there are more instances of broken anchor links broken in mui.com that we are not aware of. I wish the ahrefs SEO crawler was also reporting this. I have created https://ahrefs.canny.io/site-audit/p/html-anchor-link.

> Users do not learn anything from then compared to a line "only the pro version can paginate with more than 100 rows"

I might not agree, but fair enough.

How about we only add the h2 back (without the demo)?